Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -862,14 +862,13 @@ class BrowserViewController: UIViewController {
 
     override func observeValueForKeyPath(keyPath: String?, ofObject object: AnyObject?, change: [String: AnyObject]?, context: UnsafeMutablePointer<Void>) {
         let webView = object as! WKWebView
-        if webView !== tabManager.selectedTab?.webView {
-            return
-        }
         guard let path = keyPath else { assertionFailure("Unhandled KVO key: \(keyPath)"); return }
         switch path {
         case KVOEstimatedProgress:
-            guard let progress = change?[NSKeyValueChangeNewKey] as? Float else { break }
-            urlBar.updateProgressBar(progress)
+            if webView == tabManager.selectedTab?.webView {
+                guard let progress = change?[NSKeyValueChangeNewKey] as? Float else { break }
+                urlBar.updateProgressBar(progress)
+            }
         case KVOLoading:
             guard let loading = change?[NSKeyValueChangeNewKey] as? Bool else { break }
             toolbar?.updateReloadStatus(loading)
```

Comment:
Also, while here, please replace these two `toolbar` and `urlBar` lines to just use `navigationToolbar` instead like we do in `KVOCanGoBack`.